tgui/packages/tgui/interfaces/Canvas.tsx

```diff
@@ -148,15 +160,24 @@ class PaintCanvas extends Component<PaintCanvasProps> {
     ) {
       return;
     }
+    const coords = this.eventToCoords(event);
+    if (this.props.fillmode) {
+      this.onCanvasFill(coords.x + 1, coords.y + 1); // 1-based index dm side
+      return;
+    }
     this.modifiedElements = [];
     this.drawing = true;
     this.drawing_color = this.props.drawing_color;
-    const coords = this.eventToCoords(event);
     this.drawPoint(coords.x, coords.y, this.drawing_color);
   }
 
   drawPoint(x: number, y: number, color: any) {
-    let p: PointData = { x, y, color: Color.fromHex(color) };
+    // Checks whether an element is even
+    const isDupe = (p: PointData) => p.x === x && p.y === y;
```

Comment:
If this is a pure function it should be moved to the global scope

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-function-scoping.md